add sidekiq

dspencer001 opened the pull request.

Commit:
add sidekiq

bfcoder commented on scry.gemspec:
```diff
@@ -27,5 +27,7 @@ Gem::Specification.new do |gem|
     ["mechanize", "~> 2.7.5"],
     ["fileutils"],
+    ["sidekiq", "~> 4.2"],
+    ["thin", "~> 1.7"]
```
add a dangling comma to the end

Commit:
Add dangling comma

bfcoder reviewed (approved):
Add a comment to readme about running `redis-server`

bfcoder reviewed (commented):
So I think we need to make a decision. Do we need rack 2.0? That requires ruby 2.2.2 or above? Or do we go with rack 1.6.5 that is good with all these listed ruby versions?
If we need the new rack, then update the travis.yml
```
rvm:
  - 2.2.2
  - 2.3.1
```

bfcoder reviewed (commented):
Add `dump.rdb` to .gitignore

bfcoder commented on lib/scry/sidekiq/sidekiq.yml:
```diff
@@ -0,0 +1,3 @@
+:concurrency: 10
```
I think this would be better in the root directory.

bfcoder replied:
This file I mean.

bfcoder commented on lib/scry/sidekiq/boot.rb:
```diff
@@ -0,0 +1 @@
+Dir["./lib/scry/sidekiq/workers/*.rb"].each { |file| require file }
```
Consider adding a file in `bin/` that calls this file? So when we can `bundle exec sidekiq -r ./bin/boot.rb -C sidekiq.yml`

bfcoder commented on lib/scry/sidekiq/config.ru:
```diff
@@ -0,0 +1,8 @@
+require 'sidekiq'
```
Use double quotes

bfcoder commented on lib/scry/sidekiq/config.ru:
```diff
@@ -0,0 +1,8 @@
+  config.redis = { size: 1 }
+end
+
+require 'sidekiq/web'
```
Use double quotes

bfcoder commented on bin/monitor:
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env ruby
+
+$: << File.join(__dir__, '..', 'lib')
```
Use double quotes

bfcoder commented on lib/scry/sidekiq/workers/hard_worker.rb:
```diff
@@ -0,0 +1,10 @@
+require "sidekiq"
+
+class HardWorker
```
Should we submodule this?
```
module Scry
  class HardWorker
  end
end
```

bfcoder committed:
Add to readme
move sidekiq yml
move boot.rb
submodule hard_worker
change travis to run on ruby 2.2.2 and 2.3.1

bfcoder commented:
Looks good
